### PR TITLE
test: improve Rust runtime coverage

### PR DESCRIPTION
**Pull request by willkill07**
#### Overview

Improve Rust Runtime coverage for the native dynamic plugin adapter and other approachable gaps. The focused tests add 273 locally covered component lines, projecting the Codecov component from 94.39% to approximately 95.48%.

- [x] I confirm this contribution is my own work, or I have the right to submit it under this project's license.
- [x] I searched existing issues and open pull requests, and this does not duplicate existing work.

#### Details

- Add comprehensive native adapter coverage for ABI conversion, scope-stack lifecycle, registration entrypoints, callbacks, continuations, and stream lifecycle behavior.
- Expand dynamic plugin manifest and registry coverage for serialization, validation, duplicate records, and load-shape errors.
- Cover additional adaptive ACG profile and request-intercept paths.
- Keep the change test-only; no public API or runtime behavior changes.

Validation completed:

- `just --set ci true --set output_dir target/coverage-after test-rust`
- `RUST_TEST_THREADS=1 just test-rust`
- `cargo clippy --workspace --all-targets -- -D warnings`
- `just test-python` (496 passed)
- `just test-go`
- `just test-node` (246 passed)
- `uv run pre-commit run --all-files`

#### Where should the reviewer start?

Start with `crates/core/tests/unit/native_plugin_tests.rs`, which supplies 188 of the additional covered Rust Runtime lines and raises local native adapter coverage from 79.62% to 90.32%.

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)

- Relates to: none

## Summary by CodeRabbit

* **Tests**
  * Added unit coverage ensuring request interceptors rewrite annotations without mutating provider-facing request content.
  * Added ACG profile fingerprint tests covering alternate role sequences and seed edge cases.
  * Expanded native plugin tests for ABI boundary handling, scope stack lifecycle, error/status mapping, and streaming cancellation/cleanup.
  * Strengthened dynamic plugin manifest tests for load-contract serialization round-trips and lane-specific validation, including rejection of invalid or mixed configurations.

### CODE REVIEW

File: crates/core/tests/unit/native_plugin_tests.rs [hunk 1]
```diff
@@ -0,0 +1,1036 @@
...
+                NemoRelayNativeScopeType::Custom,
+                ptr::null(),
+                0,
+                data,
+                metadata,
+                input,
+                &timestamp,
+                &mut scope,
+            )
+        },
```
Comment (zhongxuanwang-nv) [inline crates/core/tests/unit/native_plugin_tests.rs:399]: Realized after drafting, that this is very trivial and non-blocking, please feel free to ignore!

Since these may emit events to the process-global subscriber list, should we put a `runtime_owner_test_mutex()` or `reset_global()` just like how `llm_api_tests.rs` lock_global_runtime()/reset_global() at lines 24-34) did them?

**Review by zhongxuanwang-nv — APPROVED** [review]
Everything looks great!

**Review by zhongxuanwang-nv — APPROVED** [review]
